app/controllers/api/v2/ext_app/deployments_controller.rb

```diff
@@ -0,0 +1,25 @@
+class Api::V2::ExtApp::DeploymentsController < Api::V2::ExtApp::BaseController
+  def bulk_apply
+    ActiveRecord::Base.transaction do
+      bulk_apply_params[:deployments].each do |deployment_params|
+        cluster = Cluster.find_by!(name: deployment_params.delete(:cluster_name))
+        deployment = Deployment.find_or_initialize_by(cluster: cluster, name: deployment_params[:name])
+        deployment.assign_attributes(deployment_params)
+        deployment.save!
+      end
+    end
+  end
+
+  def list_containers
```

Comment:
just realized this can also be changed to `index_containers` to make it consistent with Rails style of naming method in controller.. for routes we can probably use `GET /deployments/:id/containers`

Reply:
ack mas, i will change it to `index_containers` . But for the route path, since client only have deployment's name, we can't use `:id` as part of the path. We will use `deployments/:name/containers` instead. what do you think?

Reply:
sure that's fine using `/:name/`
just make sure name is unique and not have any blank spaces (perhaps run regex validation?)